spec.html

```diff
@@ -27557,122 +27812,85 @@ <h1>
+          1. Assert: _record_ exists, because LoadRequestedModules has completed successfully on _referrer_ prior to invoking this abstract operation.
+          1. Return _record_.[[Module]].
+        </emu-alg>
       </emu-clause>
 
-      <emu-clause id="sec-hostimportmoduledynamically" type="host-defined abstract operation">
+      <emu-clause id="sec-HostLoadImportedModule" type="host-defined abstract operation" oldids="sec-hostresolveimportedmodule,sec-hostimportmoduledynamically">
         <h1>
-          HostImportModuleDynamically (
-            _referencingScriptOrModule_: a Script Record, a Module Record, or *null*,
+          HostLoadImportedModule (
+            _referrer_: a Script Record, a Cyclic Module Record, or a Realm Record,
             _specifier_: a String,
-            _promiseCapability_: a PromiseCapability Record,
+            _hostDefined_: anything,
+            _payload_: a GraphLoadingState Record or a PromiseCapability Record,
           ): ~unused~
         </h1>
         <dl class="header">
           <dt>description</dt>
-          <dd>It performs any necessary setup work in order to make available the module corresponding to _specifier_ occurring within the context of the script or module represented by _referencingScriptOrModule_. _referencingScriptOrModule_ may be *null* if there is no active script or module when the <emu-xref href="#sec-import-calls">`import()`</emu-xref> expression occurs. It then performs FinishDynamicImport to finish the dynamic import process.</dd>
+          <dd></dd>
         </dl>
-        <p>An implementation of HostImportModuleDynamically must conform to the following requirements:</p>
 
-        <ul>
-          <li>
-            It must return ~unused~. Success or failure must instead be signaled as discussed below.
-          </li>
-          <li>
-            The host environment must conform to one of the two following sets of requirements:
-            <dl>
-              <dt>Success path</dt>
-
-              <dd>
-                <ul>
-                  <li>At some future time, the host environment must perform FinishDynamicImport(_referencingScriptOrModule_, _specifier_, _promiseCapability_, _promise_), where _promise_ is a Promise resolved with *undefined*.</li>
+        <emu-note>
+          <p>An example of when _referrer_ can be a Realm Record is in a web browser host. There, if a user clicks on a control given by</p>
 
-                  <li>Any subsequent call to HostResolveImportedModule after FinishDynamicImport has completed, given the arguments _referencingScriptOrModule_ and _specifier_, must return a normal completion containing a module which has already been evaluated, i.e. whose Evaluate concrete method has already been called and returned a normal completion.</li>
-                </ul>
-              </dd>
+          <pre><code class="html">&lt;button type="button" onclick="import('./foo.mjs')"&gt;Click me&lt;/button&gt;</code></pre>
```

Comment:
Frankly it's pretty weird that this has `null` active script. (For reference, it's in step 9 [here](https://html.spec.whatwg.org/#internal-raw-uncompiled-handler).)

Not something to worry about here, though.

Reply:
Maybe we can eventually get HTML to create a script for inline handlers :shrug: